nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -531,6 +532,8 @@ FAIL:
 			if res.createEval || len(res.allowReplacements) != 0 {
 				w.createBatchedUpdate(res.allowReplacements, allocIndex)
 			}
+
+			allocsCh = w.getAllocsCh(allocIndex)
```

Comment:
Ideally this line is immediately after the `allocIndex` but I wanted to avoid creating a query in error or failure conditions. Though, given that those conditions are relatively rare, I can be persuaded to move it if it improves readability.